壊れたリンクを修正する

1. エラー原因: Docusaurus で壊れたリンクが見つかりました。
2. 対処方法: 該当ページのリンクを確認し、存在しないパスを参照していないか確認してください。

## 壊れたリンクの一覧

1. ソースページパス = /reference/single-process-and-callback:
   - promise-async-await.md へのリンク
2. ソースページパス = /reference/values-types-variables/tuple:
   - ../promise-async-await.md へのリンク
3. ソースページパス = /tutorials/nextjs:
   - /reference/promise-async-await#async%E9%96%A2%E6%95%B0 へのリンク
   - /reference/promise-async-await#await へのリンク

## エラー内容

```
[ERROR] Unable to build website for locale ja.
[ERROR] Error: Docusaurus found broken links!

Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.

Exhaustive list of all broken links found:

- On source page path = /reference/single-process-and-callback:
   -> linking to promise-async-await.md (resolved as: /reference/promise-async-await.md)

- On source page path = /reference/values-types-variables/tuple:
   -> linking to ../promise-async-await.md (resolved as: /reference/promise-async-await.md)

- On source page path = /tutorials/nextjs:
   -> linking to /reference/promise-async-await#async%E9%96%A2%E6%95%B0 (resolved as: /reference/promise-async-await)
   -> linking to /reference/promise-async-await#await (resolved as: /reference/promise-async-await)

    at reportMessage (/Volumes/x/yytypescript/book/node_modules/@docusaurus/utils/lib/index.js:306:19)
    at handleBrokenLinks (/Volumes/x/yytypescript/book/node_modules/@docusaurus/core/lib/server/brokenLinks.js:138:35)
    at async buildLocale (/Volumes/x/yytypescript/book/node_modules/@docusaurus/core/lib/commands/build.js:155:5)
    at async tryToBuildLocale (/Volumes/x/yytypescript/book/node_modules/@docusaurus/core/lib/commands/build.js:33:20)
    at async mapAsyncSequencial (/Volumes/x/yytypescript/book/node_modules/@docusaurus/utils/lib/index.js:262:24)
    at async build (/Volumes/x/yytypescript/book/node_modules/@docusaurus/core/lib/commands/build.js:68:25)
error Command failed with exit code 1.
```

## 再現方法

- ローカルに c8e98459e1628753e0880c1c19645cc7f15cc109 をチェックアウト
- yarn build を実行

## ビルドがFailになっているコミット

- c8e98459e1628753e0880c1c19645cc7f15cc109

## 追加対応(オプション)

- GitHub Actionsで `yarn build` するワークフローを追加してCIで検出できるようにする

--- a/docs/reference/single-process-and-callback.md
+++ b/docs/reference/single-process-and-callback.md
@@ -198,4 +198,4 @@ ajax("https://...", (res1: Response) => {
 
 このコールバック地獄を解消する画期的なクラスとして`Promise`が登場し主要なブラウザとNode.jsではビルトインオブジェクトとして使うことができます。こちらの説明については本書に専用のページがありますのでそちらをご参照ください。
 
-[Promise / async / await](promise-async-await.md)
+[非同期処理](./asynchronous/README.md)

--- a/docs/reference/values-types-variables/tuple.md
+++ b/docs/reference/values-types-variables/tuple.md
@@ -101,7 +101,7 @@ const [, , bool]: [number, string, boolean] = tuple();
 TypeScriptで非同期プログラミングをする時に、時間のかかる処理を直列ではなく並列で行いたい時があります。そのときTypeScriptでは`Promise.all()`というものを使用します。このときタプルが役に立ちます。
 `Promise`についての詳しい説明は本書に専門の頁がありますので譲ります。ここでは`Promise<T>`という型の変数は`await`をその前につけると`T`が取り出せることだけ覚えておいてください。また、この`T`をジェネリクスと言いますが、こちらも専門の頁があるので譲ります。
 
-[Promise / async / await](../promise-async-await.md)
+[非同期処理](../asynchronous/README.md)
 
 [ジェネリクス (generics)](/reference/generics)
 

--- a/docs/tutorials/nextjs.md
+++ b/docs/tutorials/nextjs.md
@@ -171,7 +171,7 @@ const fetchImage = async () => {
 
 `fetch`はHTTPリクエストでリソースを取得するブラウザ標準のAPIです。戻り値として[Response](https://developer.mozilla.org/ja/docs/Web/API/Response)オブジェクトを返します。Responseオブジェクトの`json()`メソッドを実行することで、レスポンスのボディーをJSONとしてパースし、JavaScriptのオブジェクトとして取得できます。
 
-`fetchImage`関数についている[`async`キーワード](/reference/promise-async-await#async関数)は、この関数が非同期処理を行うことを示すものです。`fetch`と`res.json`は非同期関数で、これらの処理を待つために、それぞれに[`await`キーワード](/reference/promise-async-await#await)がついています。
+`fetchImage`関数についている[`async`キーワード](/reference/asynchronous/async)は、この関数が非同期処理を行うことを示すものです。`fetch`と`res.json`は非同期関数で、これらの処理を待つために、それぞれに[`await`キーワード](/reference/asynchronous/await)がついています。
 
 <!-- todo: async/awaitのリライトが完了したらリンク先を変更する -->
 

